assume/strategies/learning_advanced_orders.py

```diff
@@ -264,17 +265,34 @@ def create_observation(
         # =============================================================================
         # 1.1 Get the Observations, which are the basis of the action decision
         # =============================================================================
-        scaling_factor_res_load = self.max_demand
 
+        # defines bounds of observation space
+        # stays here as it is unit specific, and different forecasts might apply for different units
+        # different handling would require an extra unit loop at learning role intiliazation and unit specific max/min values
+        # further forecasts might change during the simulation if advanced forecasting is used
+        self.max_market_price = max(unit.forecaster[f"price_{market_id}"])
+        self.min_market_price = min(unit.forecaster[f"price_{market_id}"])
+        self.max_residual = max(unit.forecaster[f"residualy_load_{market_id}"])
+        self.min_residual = min(unit.forecaster[f"residual_load_{market_id}"])
+
+        # residual load forecast
+        upper_scaling_factor_res_load = self.max_residual
+        lower_scaling_factor_res_load = self.min_residual
+        
         # price forecast
-        scaling_factor_price = self.max_bid_price
+        upper_scaling_factor_price = self.max_market_price
```

Comment:
why assign new variable here if we can use self.values directly?

Reply:
We did the same beforehand to make this crucial part of the learning easily changeable for new users; compare the RL tutorial. We could change it if you think it would be easier to understand.